classes/check/tasklatencycheck.php

```diff
@@ -0,0 +1,139 @@
+<?php
+// This file is part of Moodle - http://moodle.org/
+//
+// Moodle is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Moodle is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Moodle.  If not, see <http://www.gnu.org/licenses/>.
+/**
+ * Individual cron task latency monitoring check check.
+ *
+ * @package    tool_heartbeat
+ * @copyright  2022
+ * @author     Peter Burnett <peterburnett@catalyst-au.net>
+ * @license    http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ *
+ * This can be run either as a web api, or on the CLI. When run on the
+ * CLI it conforms to the Nagios plugin standard.
+ *
+ * See also:
+ *  - http://nagios.sourceforge.net/docs/3_0/pluginapi.html
+ *  - https://nagios-plugins.org/doc/guidelines.html#PLUGOUTPUT
+ */
+
+namespace tool_heartbeat\check;
+use core\check\check;
+use core\check\result;
+
+defined('MOODLE_INTERNAL') || die();
+require_once("$CFG->libdir/cronlib.php");
+
+class tasklatencycheck extends check {
+
+    /**
+     * Get the status of the monitored tasks.
+     *
+     * @return result
+     */
+    public function get_result(): result {
+        global $CFG, $DB;
+
+        $taskconfig = get_config('tool_heartbeat', 'tasklatencymonitoring');
+        $status = result::OK;
+        $messages = '';
+
+        $lockstats = isset($CFG->lock_factory) && $CFG->lock_factory === '\\tool_lockstats\\proxy_lock_factory';
```

Comment:
I've always found the `:class` syntax a bit nicer (+ has syntax highlighting and you can jump to the definition if needed).
```suggestion
        $lockstats = isset($CFG->lock_factory) && $CFG->lock_factory === \tool_lockstats\proxy_lock_factory::class;
```
What do you think?

Reply:
If the class doesnt exist then this will die for things like lockstats checks when not installed. Ill see what I can swap

Reply:
It actually won't. :rocket:

Reply:
Was only able to swap the guaranteed existing classes in the  tests, the rest isn't safe